pkg/api-server/server.go

```diff
@@ -117,47 +116,75 @@ func NewApiServer(
 		AllowedDomains: serverConfig.CorsAllowedDomains,
 		Container:      container,
 	}
+	container.Filter(cors.Filter)
 
 	// We create a WebService and set up resources endpoints and index endpoint instead of creating WebService
 	// for every resource like /meshes/{mesh}/traffic-permissions, /meshes/{mesh}/traffic-log etc.
 	// because go-restful detects it as a clash (you cannot register 2 WebServices with path /meshes/)
 	ws := new(restful.WebService)
 	ws.
-		Path("/").
+		Path(cfg.ApiServer.BasePath).
 		Consumes(restful.MIME_JSON).
 		Produces(restful.MIME_JSON)
 
 	addResourcesEndpoints(ws, defs, resManager, cfg, access.ResourceAccess)
 	addPoliciesWsEndpoints(ws, cfg.Mode, cfg.ApiServer.ReadOnly, defs)
 	addInspectEndpoints(ws, cfg, meshContextBuilder, resManager)
 	addInspectEnvoyAdminEndpoints(ws, cfg, resManager, access.EnvoyAdminAccess, envoyAdminClient)
-	container.Add(ws)
-
-	if err := addIndexWsEndpoints(ws, getInstanceId, getClusterId, enableGUI); err != nil {
-		return nil, errors.Wrap(err, "could not create index webservice")
+	addZoneEndpoints(ws, resManager)
+	guiUrl := cfg.ApiServer.GUI.BasePath
+	if cfg.ApiServer.GUI.RootUrl != "" {
+		guiUrl = cfg.ApiServer.GUI.RootUrl
+	}
+	apiUrl := cfg.ApiServer.BasePath
+	if cfg.ApiServer.RootUrl != "" {
+		apiUrl = cfg.ApiServer.RootUrl
 	}
-	configWs, err := configWs(cfg)
+
+	err := addConfigEndpoints(ws, cfg)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not create configuration webservice")
 	}
-	container.Add(configWs)
-	container.Add(zonesWs(resManager))
-	container.Add(tokenWs(tokenIssuers, access))
+	enableGUI := cfg.ApiServer.GUI.Enabled && cfg.Mode != config_core.Zone
```

Comment:
Should we warn on people trying to enable GUI on zone CP?

Reply:
The trick here is that the GUI is enabled by default in the conf. I'm going to add a bit more info in the error page